### PR TITLE
Update post

### PR DESCRIPTION
**Pull request by ajosephides**
A user can update a post.
They can only do so within 10 mins of creation.
They must be the post owner.
NOTE: time constraints have meant it is a little 'hacky' but it is tested.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -39,6 +39,7 @@ group :development, :test do
   gem 'rubocop', require: false
   gem 'selenium-webdriver'
   gem 'simplecov', require: false
+  gem 'timecop'
```
Comment (IndecentDolphin) [inline Gemfile]: Extremely useful

File: spec/features/user_update_post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    expect(page).to have_content("Error: you can\'t update other users messages")
+  end
+end
+
+RSpec.feature "User can update a post", type: :feature do
+  scenario "User can update a post within 10 mins of creation" do
+    user = User.create(name:"Jean-Claude", surname:"Van Damme", email:"doublehard@test.com", password:"1234567")
+    post = user.posts.create(message:"All my plumbing is coconut water")
+    new_time = Time.now.utc + 11.minutes
+    Timecop.travel(new_time)
```
Comment (l-palermo) [inline spec/features/user_update_post_spec.rb]: What time cop travel do?

File: spec/features/user_update_post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    visit '/'
+    fill_in 'session_email', with: 'doublehard@test.com', visible: false
+    fill_in 'session_password', with: '1234567', visible: false
+    click_button 'Log in'
+    click_link 'All my plumbing is coconut water'
+    click_link 'Edit'
+    fill_in "Message", with: "You see, I'm an ambitious Harvard-educated visionary who deserves to be the most powerful man in the world"
+    click_button 'Update'
+    expect(page).to have_content("Error: you can\'t update a post since it has been 10 minutes since it was created")
+    Timecop.return
```
Comment (l-palermo) [inline spec/features/user_update_post_spec.rb]: And this one?

File: spec/features/user_update_post_spec.rb (hunk 2 above)
Comment (ajosephides) [reply spec/features/user_update_post_spec.rb]: It allows you to mock time.

File: spec/features/user_update_post_spec.rb (hunk 3 above)
Comment (ajosephides) [reply spec/features/user_update_post_spec.rb]: Timecop works by mocking the system time. You need to reset it back to normal.
Docs are here:
https://github.com/travisjeffery/timecop

**Pull request closed without merging**